### DISCUSSION TITLE
Quotation marks on lemmatization after read file with Pandas

### DISCUSSION THREAD

**Discussion by k4waiiiii** [explosion/spaCy#9768] [Help: Coding & Implementations] [answered]
I have tokenization and lemmatization then saved it to a file called 'data after lemmatization.xlsx' 

This is the data before saved it to file:
![before saved](https://user-images.githubusercontent.com/49643347/143896044-ea087538-3424-440e-8086-0902ee2e72f5.png)

This is the data after saved it to file with Pandas and then read the data file using Pandas:
![after saved and read the data](https://user-images.githubusercontent.com/49643347/143896506-9ed52a27-32dc-4737-91ad-67bc4be6eaa0.png)

Why I have quotation marks on column `lemmatization` but on column `tokenizing` is not?
```python
import pandas as pd
from spacy.lang.id import Indonesian

nlp = Indonesian()
nlp.add_pipe('lemmatizer')
nlp.initialize()

data = [
    'pergi untuk melakukan penanganan banjir'
]

df = pd.DataFrame({'text': data})

#Tokenization
def tokenizer(words):
    return [token for token in nlp(words)]

#Lemmatization
def lemmatizer(token):
    return [lem.lemma_ for lem in token]


df['tokenizing'] = df['text'].apply(tokenizer)
df['lemmatizaton'] = df['tokenizing'].apply(lemmatizer)
df.to_excel('./data after lemmatization.xlsx', index=False)

dc = pd.read_excel('./data after lemmatization.xlsx')
dc
```

**Comment by polm** [comment, marked as answer]
It looks like the contents of your `tokenizing` column are Tokens, while the contents of your `lemmatization` column are strings. These are displayed differently because of the way `repr` works in Python - it renders strings with single quotes, and spaCy tokens just print the text.

You can make them the same by converting the spaCy tokens to strings.

Note this isn't really a spaCy specific issue.

**Reply by k4waiiiii** [reply]
How converting the spaCy tokens to strings? Trying with `token.text` but got an error on `lemma_` the error is `AttributeError: 'str' object has no attribute 'lemma_'`

**Reply by polm** [reply]
`token.text` is the right way to get a string from a Token. But strings don't have lemmas, like the error says. You have to:

1. Get the tokens
2. Get the lemmas from the tokens
3. Convert the tokens to strings

It seems like what you are currently doing is:

1. Get the tokens
2. Convert the tokens to strings
3. Get the lemmas from the strings (won't work)

**Reply by k4waiiiii** [reply]
Ok. I see. Thanks for answering.
